templates/collection.liquid

```diff
@@ -1,9 +1,93 @@
+<section id="collection">
+  <div class="container">
+    <div class="grid-row">
+
+{% paginate collection.products by 24 %}
+
 {% if collection.all_products_count > 0 %}
-{% for product in collection.products %}
-<h2><a href="{{ product.url }}">{{ product.title }}</a></h2>
-{{ product.description }}
-<img src="{{ product.featured_image | product_img_url: 'grande' }}" alt="{{ product.title | escape }}" />
-{% endfor %}
+
+  {% if collection.url != blank %}
+
+      <div class="grid-item item-s-12 item-m-auto grid-row no-gutter align-content-start">
+        <div class="grid-item item-s-6 item-m-12 margin-bottom-tiny">
+          {% comment %}Types List{% endcomment %}
+          <h3 class="font-bold font-uppercase font-size-small">Type</h3>
+          <ul>
+          {% for product_type in collection.all_types %}
+          <li>
+            <a href="{{ collection.url }}/{{ product_type | handle }}">{{ product_type }}</a>
+          </li>
+          {% endfor %}
+          </ul>
+        </div>
+        {% comment %}Brands List{% endcomment %}
+        <div class="grid-item item-s-6 item-m-12">
+          <h3 class="font-bold font-uppercase font-size-small">Brand</h3>
+          <ul>
+          {% for product_vendor in collection.all_vendors %}
+            <li>
+          		<a href="{{ collection.url }}/{{ product_vendor | handle }}">{{ product_vendor }}</a>
+            </li>
+          {% endfor %}
+          </ul>
+        </div>
+      </div>
+
+  {% endif %}
+
+      <div class="grid-item flex-grow no-gutter grid-row">
+
+        {% comment %}Breadcrumbs{% endcomment %}
+        <div class="grid-item item-s-12 font-bold font-size-small font-uppercase">
+          {% if current_tags != blank %}
+          <a href="{{ collection.url }}">{{ collection.handle }}</a> > {{ current_tags }}
+          {% else %}
+          {{ collection.handle }}
+          {% endif %}
+        </div>
+
+  {% for product in collection.products %}
+        {% comment %}Product{% endcomment %}
+        <div class="grid-item item-s-6 item-l-4 item-xl-3 font-size-small">
+
+          <a href="{{ product.url }}">
+            <div class="text-align-center"><img src="{{ product.featured_image | product_img_url: 'large' }}" alt="{{ product.title | escape }}" /></div>
```

Comment:
Is there any way to do srcset with Shopify?

Reply:
```
{% for product in collection.products %}
  <a href="{{ product.url | within: collection }}">
    {% assign image = product.featured_image %}
    <img src="{{ image.src | img_url: '480x' }}" alt="{{ image.alt | escape }}"
      srcset="
      {% if image.width > 640 %}{{ image.src | img_url: '640x' }} 640w{% endif %}
      {% if image.width > 720 %},{{ image.src | img_url: '720x' }} 720w{% endif %}
      {% if image.width > 900 %},{{ image.src | img_url: '900x' }} 900w{% endif %} 
      " 
      sizes="(min-width: 960px) 450px, (min-width: 720px) 50vw, 100vw"
    />
  </a>
{% endfor %}
```

Reply:
https://www.shopify.com/partners/blog/the-img_url-filter-just-got-10x-better